types/datasource/proctree.go

```diff
@@ -0,0 +1,80 @@
+package datasource
+
+import "time"
+
+// ProcessInfo is the user facing representation of a process data at a specific time.
+type ProcessInfo struct {
+	Hash              uint32
+	Pid               int
+	NsPid             int
+	Ppid              int
+	ContainerId       string
+	Cmd               []string
+	ExecutionBinary   FileInfo
+	Interpreter       FileInfo
+	Interp            FileInfo
+	StartTime         time.Time // First thread fork time. TODO: Can we use the main thread start time instead?
+	ExecTime          time.Time // Last execve call time
+	ExitTime          time.Time
+	ParentHash        uint32
+	ThreadsIds        map[int]uint32 // Map between the ids to their hashes
+	ChildProcessesIds map[int]uint32 // Map between the ids to their hashes
+	IsAlive           bool
+}
+
+// ThreadInfo is the user facing representation of a thread data at a specific time.
+type ThreadInfo struct {
+	Hash        uint32
+	Tid         int
+	NsTid       int
+	Pid         int
+	UserId      int
+	GroupId     int
+	ForkTime    time.Time
+	ExitTime    time.Time
+	Name        string
+	ProcessHash uint32
```

Comment:
Is this hash from the "thread group leader" ? Maybe we should use the same names both sides (process tree and here).

Reply:
So this connects to my previous comment.
There should be a clear distinction between processes and threads.
I think we have too much of a mish-mash between kernel terms and user terms.
This should be the process hash, because the user is interested in the process information. Not necessarily the thread group leader information (the `TaskInfo` of the group leader is irrelevant for most use cases of the process  tree). 
* I took a closer look on the current design, and I see that some crucial process information is accessible only through the `TaskInfo` struct of the process. We need to discuss the design here to think what is the best API here.

Reply:
Agree, just need to decide on the name

Reply:
I believe in the end that having more user-friendly names here is more appropriate.
Tell me what you think.